Program.cs

```diff
@@ -1,10 +1,20 @@
-﻿namespace Training
-{
-    internal class Program
-    {
-        static void Main(string[] args)
-        {
-            Console.WriteLine("Hello, World!");
-        }
-    }
-}
\ No newline at end of file
+﻿/*  Check if the paswword is strong or not.
+The password must contain atleast 6 characters.
+The password must contain atleast one number.
+The password must contain atleast one special character.
+The password must contain atleast one uppercase and lowercase letters. */
+
+using System.Text;
+
+Console.WriteLine ("Enter your password: ");
+string password = Console.ReadLine ();
+string specialCharacter = "!@#$%^&*()-+";
+if (password.Length >= 6) {
+   StringBuilder weakReason = new ("Your password must contain atleast ");
+   if (!password.Any (char.IsDigit)) weakReason.Append ("one digit, ");
+   if (!password.Any (char.IsUpper)) weakReason.Append ("one uppercase letter, ");
+   if (!password.Any (char.IsLower)) weakReason.Append ("one lowercase letter, ");
+   if (!password.Any (x => specialCharacter.Contains (x))) weakReason.Append ("one special character !@#$%^&*()-+");

```

Comment:
Why do we need a x variable inside any. String itself a collection of characters only. You should reuse the special character string in the message do not hard code it again. Say if you are adding/removing some special character you need to update in two places which is not a good practice.